### PR TITLE
fix: don't add guide properties to tooltips, use encoding titles for boxplot titles

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

File: src/compositemark/common.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ export function filterTooltipWithAggregatedField<F extends Field>(
 } {
   const {tooltip, ...filteredEncoding} = oldEncoding;
   if (!tooltip) {
-    return {filteredEncoding: oldEncoding};
+    return {filteredEncoding};
```
Comment (kanitw) [inline src/compositemark/common.ts:61]: Why this this change related to "fix: don't add guide properties to tooltips"?

File: src/compositemark/common.ts [hunk 2]
```diff
@@ -123,11 +123,16 @@ export function getCompositeMarkTooltip(
     }
   );
 
+  const tooltipFieldDefs = fieldDefs(encodingWithoutContinuousAxis).map(fd =>
+    // remove guides as they don't work for tooltips
+    omit(fd, ['legend', 'axis', 'header'] as any[])
```
Comment (kanitw) [inline src/compositemark/common.ts:128]: What about scale?  

I also look at `tooltip.ts`, it seems like our tooltip logic reads `title` for its key.  So simply drop guides here is also incorrect.

File: src/compositemark/common.ts (hunk 1 above)
Comment (domoritz) [reply src/compositemark/common.ts:61]: It's not. It just makes the code less confusing.

File: src/compositemark/common.ts (hunk 2 above)
Comment (domoritz) [reply src/compositemark/common.ts:128]: Oh, we read the title off the guide? We should move the title to the encoding property then.

Comment (kanitw) [reply src/compositemark/common.ts:128]: Yeah, I'd say look at tooltip.ts carefully and make sure we keep all important bits.

File: src/channeldef.ts [hunk 3]
```diff
@@ -885,8 +897,8 @@ export function title(
   config: Config,
   {allowDisabling, includeDefault = true}: {allowDisabling: boolean; includeDefault?: boolean}
 ) {
-  const guide = getGuide(fieldOrDatumDef) ?? {};
-  const guideTitle = guide.title;
+  const guide = getGuide(fieldOrDatumDef);
```
Comment (kanitw) [inline src/channeldef.ts:900]: We can get rid of guide variable?

File: src/compositemark/boxplot.ts [hunk 4]
```diff
@@ -291,7 +291,7 @@ export function normalizeBoxPlot(
           type: continuousAxisChannelDef.type,
           ...(title !== undefined ? {title} : {}),
           ...(scale !== undefined ? {scale} : {}),
-          ...(axis !== undefined ? {axis} : {})
+          ...(isEmpty(axisWithoutTitle) ? {} : {axis: axisWithoutTitle})
```
Comment (kanitw) [inline src/compositemark/boxplot.ts:294]: Add comment why we need `axisWithoutTitle` here.  (It's unclear to me.)

File: src/compositemark/common.ts [hunk 5]
```diff
@@ -171,7 +175,7 @@ export function makeCompositeAggregatePartFactory<P extends PartsMixins<any>>(
           type: continuousAxisChannelDef.type,
           ...(title !== undefined ? {title} : {}),
           ...(scale !== undefined ? {scale} : {}),
-          ...(axis !== undefined ? {axis} : {})
+          ...(isEmpty(axisWithoutTitle) ? {} : {axis: axisWithoutTitle})
```
Comment (kanitw) [inline src/compositemark/common.ts:178]: Add comment why we need axisWithoutTitle here. (It's unclear to me.)

**Pull request merged**